ui/views.py

```diff
@@ -90,60 +96,77 @@ def get_context_data(self, **kwargs):
         return context
 
 
-@require_POST
-@login_required
-@admin_required
-def stream(request):
-    """stream"""
-    data = json.loads(request.body.decode('utf-8'))
-    serializer = DropboxFileSerializer(data=data, many=True)
-    serializer.is_valid(raise_exception=True)
-    dropbox_files = serializer.validated_data
-    response = {}
-    for dropbox_file in dropbox_files:
-        video = Video.objects.create(
-            source_url=dropbox_file["link"],
-            creator=request.user,
-            title=dropbox_file["name"]
-        )
-        try:
-            VideoFile.objects.create(
-                s3_object_key=video.s3_key(),
-                video_id=video.id,
-                bucket_name=settings.VIDEO_S3_BUCKET
-            )
-        except Exception as e:
-            # Delete the video object if videofile object can't be created, no point in keeping it around
-            video.delete()
-            raise e
-        # Kick off chained async celery tasks to transfer file to S3, then start a transcode job
-        task_result = chain(stream_to_s3.s(dropbox_file["link"], video.s3_key()), transcode_from_s3.si(video.id))()
-        response[video.id] = {
-            "key": video.s3_key(),
-            "task": task_result.id,
-        }
-
-    return JsonResponse(response)
+class UploadVideosFromDropbox(APIView):
+    """
+    Class based view for uploading videos from dropbox to S3.
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+    )
+    permission_classes = (
+        permissions.IsAuthenticated,
```

Comment:
The original upload view required admin permissions but I think just being an authenticated user makes more sense.